nav2_mppi_controller/src/critics/goal_angle_critic.cpp

```diff
@@ -41,16 +41,15 @@ void GoalAngleCritic::score(CriticData & data)
     return;
   }
 
-  const auto goal_idx = data.path.x.shape(0) - 1;
+  const auto goal_idx = data.path.x.size() - 1;
   const float goal_yaw = data.path.yaws(goal_idx);
 
-  if (power_ > 1u) {
-    data.costs += xt::pow(
-      xt::mean(xt::fabs(utils::shortest_angular_distance(data.trajectories.yaws, goal_yaw)), {1}) *
-      weight_, power_);
+  if(power_ > 1u) {
+    data.costs += (((utils::shortest_angular_distance(data.trajectories.yaws, goal_yaw).abs()).
```

Comment:
For all: is there a `fabs` option to not case into double?

Reply:
Eigen itself is handling it, they have defined overloaded abs() for both float and double. 
![Screenshot from 2024-10-17 01-59-42](https://github.com/user-attachments/assets/47a0e751-8d18-430c-bd6c-192312eb9d75)